src/execution/compiler/codegen.cpp

```diff
@@ -182,6 +188,8 @@ ast::Expr *CodeGen::Int32Type() const { return BuiltinType(ast::BuiltinType::Int
 
 ast::Expr *CodeGen::Int64Type() const { return BuiltinType(ast::BuiltinType::Int64); }
 
+ast::Expr *CodeGen::Uint32Type() const { return BuiltinType(ast::BuiltinType::Uint32); }
```

Comment:
So I think I probably should have asked this while developing, instead of waiting for the review, but why isn't there already an unsigned integer type? I think I honestly might just be missing or misunderstanding something, I'm pretty sure TPL works with unsigned integers. The reason I added this was for column oids, which aren't unique to ANALYZE. Have we just casted to and from signed integers?

Reply:
Sorry, let me just answer my own question instead of being lazy. Here in the update translator we create an array of column oids (It's what I based the analyze translator code off of that uses TPL unsigned ints):
https://github.com/cmu-db/noisepage/blob/7c79856e4aa242c3747ddd642bec21fd9535d5f7/src/execution/compiler/operator/update_translator.cpp#L149-L160
Specifically this line `ast::Expr *rhs = GetCodeGen()->Const32(all_oids_[i].UnderlyingValue());` 
`all_oids_` is a vector of `col_oid_t`s which is an alias for `uint32_t`. Updates clearly work, but why are we using signed integers in TPL instead of unsigned? I don't really know.

Reply:
Copying what Prashanth said on Slack since it's relevant
> There was never a need for unsigned constant integers during SQL processing.
The update translator logic is wrong. But, it works because the type-checker figures out the constants are all positive anyways.
Feel free to add unsigned versions to the `CodeGen` class.
TPL supports both signed and unsigned integers.
Check out upstream [here](https://github.com/pmenon/tpl/tree/icg) to see if the problem you're having in master has been resolved already. No need to reinvent things.

So it seems like I should add an unsigned const function to codegen. As far as naming goes I'm really open to anything.

Reply:
Is this something you want to bring in with ANALYZE? Or do you think we should:

- Add an issue for it
- Open a separate PR to address it

?

Reply:
Let's open up a separate issue about it, we should probably take a look at the upstream and base any change off that.